src/System.Text.RegularExpressions/ref/System.Text.RegularExpressions.cs

```diff
@@ -62,9 +62,22 @@ public partial class MatchCollection : System.Collections.ICollection, System.Co
         void System.Collections.ICollection.CopyTo(System.Array array, int arrayIndex) { }
     }
     public delegate string MatchEvaluator(System.Text.RegularExpressions.Match match);
+    abstract public partial class RegexRunnerFactory
+    {
+        protected RegexRunnerFactory() { }
+        abstract protected internal RegexRunner CreateInstance();
+    }
     public partial class Regex
     {
         public static readonly System.TimeSpan InfiniteMatchTimeout;
+        protected internal string pattern;
```

Comment:
What is the advantage of directly exposing these fields both here and in RegexRunner?

Wouldn't a better pattern be to make the fields private, and have properties that are `protected internal` (and PascalCased) that return/set the values?